### PR TITLE
React & Redux group project - Space Travelers' Hub

### PR DESCRIPTION
**Pull request by aliveGUY**
# React & Redux group project - Space Travelers' Hub :rocket: 
### Config & basic setup
:heavy_check_mark:  Create React App.
:heavy_check_mark:  Install React Redux Redux Logger and React Router.
:heavy_check_mark:  Download the free image for the app logo.
:heavy_check_mark:  Create routes and view components (Rockets, Missions, My Profile, Dragons [only if your team has 3 members]). Use <NavLink /> for the page navigation links and style active class to indicate which section/page user is currently on (underline active navigation link).
:heavy_check_mark:  Create directories for all Redux state slice files (rockets, missions, dragons [only if your team has 3 members]).
### Redux: Fetch data and update the Redux store
:heavy_check_mark:  Upon first render fetch data from the SpaceX API endpoints
### Render UI:lists
:heavy_check_mark:  Use useSelector() Redux Hook to select the state slices and render lists of rockets and missions in corresponding routes. i.e.:
:heavy_check_mark:  Render a list of rockets (as per design). For the image of a rocket use the first image in the array of flickr_images.
:heavy_check_mark:  Render a table with the missions' data (as per design).
### Redux: Write actions and reducers for booking rockets/dragons and joining missions
:heavy_check_mark:  Create a reducer and action dispatcher for the "Join Mission" button. The logic here is practically the same as with rockets - you need to pass the mission's ID to the corresponding action and update the missions' state with the selected mission having a new key/value - `reserved: true`
### Render UI: conditional components rendering
:heavy_check_mark:  Rockets that have already been reserved should show a "Reserved" badge and "Cancel reservation" button instead of the default "Reserve rocket" (as per design) .
:heavy_check_mark:  Missions that the user has joined already should show a badge "Active Member" instead of the default "NOT A MEMBER" and a button "Leave Mission" instead of the "Join Mission" button (as per design).
### Technical set up**
:heavy_check_mark:  Set up the repository on GitHub and use Gitflow.
:heavy_check_mark:  Set up React and Redux.
:heavy_check_mark:  Set up testing libraries(React Testing Library and Jest).
:heavy_check_mark:  Do not preserve user data.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    dispatch(dispatchRockets());
+  }, []);
+
+  return (
+    <Router>
+      <Nav />
+      <Routes>
+        <Route exact path="/" element={<Missions />} />
+        <Route exact path="/myprofile" element={<MyProfile />} />
+        <Route exact path="/rockets" element={<Rockets />} />
```
Comment (mateo951) [inline src/App.js:29]: Proper routes and view components are created properly 🙌

File: src/index.css [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+}
+
+.list-group-item {
+  width: 70%;
+  padding-right: 235px;
+}
+
+.active {
+  text-decoration: underline;
+}
```
Comment (mateo951) [inline src/index.css:149]: The active page should be indicated with an underline ✔

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import './index.css';
+
+const root = ReactDOM.createRoot(document.getElementById('root'));
+root.render(
+  <React.StrictMode>
+    <ThemeProvider
+      breakpoints={['xxxl', 'xxl', 'xl', 'lg', 'md', 'sm', 'xs', 'xxs']}
+      minBreakpoint="xxs"
+    >
+      <Provider store={store}>
```
Comment (mateo951) [inline src/index.js:17]: Redux store is configured properly! Well done 🔥

File: src/redux/missions/Redux-Missions.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+const url = 'https://api.spacexdata.com/v3/missions';
```
Comment (mateo951) [inline src/redux/missions/Redux-Missions.js:1]: Rockets and Missions data is being fetched successfully from SpaceX API 🤓

File: src/components/MyProfile.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import ListGroup from 'react-bootstrap/ListGroup';
+
+export default function MyProfile() {
+  const data = useSelector((state) => state.missions);
```
Comment (mateo951) [inline src/components/MyProfile.js:6]: useSelector() is used to get a state slice corresponding to routes ✔

File: src/components/Missions.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+                </Badge>
+              </th>
+              <th className="vert">
+                <Button
+                  variant={`${mission.reserved ? 'outline-danger' : 'outline-secondary'}`}
+                  className="m-3"
+                  type="button"
+                  onClick={() => {
+                    dispatch(reserveMission(mission.mission_id));
+                  }}
```
Comment (mateo951) [inline src/components/Missions.js:35]: "Join Mission" button updates the store successfully by dispatching an action 💪😎 Well done ✔

File: src/components/Rockets.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+            </Col>
+            <Col sm={8}>
+              <p className="h4">{rocket.name}</p>
+              {rocket.reserved && (
+                <Button variant="info" className="pr-5 badge mx-1">Reserved</Button>
+              )}
+
+              <p className="d-inline">{rocket.description}</p>
+              <br />
+              {rocket.reserved ? <Button className="mt-3" variant="outline-secondary" size="md" onClick={() => dispatch(reserveRocket(rocket.id))}>Cancel Reservation Rocket</Button> : <Button variant="primary" size="md" className="mt-3" onClick={() => dispatch(reserveRocket(rocket.id))}>Reserve Rocket</Button> }
```
Comment (mateo951) [inline src/components/Rockets.js:27]: "Reserve Rocket" button updates the store successfully by dispatching an action 🤯 Well done guys ✔

File: src/components/MyProfile.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  const missions = data.filter((mission) => mission.reserved);
+  const reservedRockets = useSelector((state) => state.rockets).filter((rocket) => rocket.reserved);
+  return (
```
Comment (mateo951) [inline src/components/MyProfile.js:9]: Awesome 🔥 You are rendering a list of reserved rockets and missions without any issue!

File: src/components/MyProfile.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import ListGroup from 'react-bootstrap/ListGroup';
+
+export default function MyProfile() {
+  const data = useSelector((state) => state.missions);
+  const missions = data.filter((mission) => mission.reserved);
+  const reservedRockets = useSelector((state) => state.rockets).filter((rocket) => rocket.reserved);
```
Comment (mateo951) [inline src/components/MyProfile.js:8]: The filter() method is used to render the reserved rockets and missions 🚀 Fantastic

**Review by mateo951 — APPROVED** [review]
### Hi Team, 

📝👌 **Your project highlights** 😎 
- General Requirements 💪🤓
- Gitflow (Reviews for pair-programming) 🙌
- Data fetch from API's 🤯
- Rendering components properly 🔥

You did an excellent job 🤓 You should be very proud of yourselves 😎

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 
### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by aliveGUY** [comment]
@wandergithub

**Pull request merged**